### DISCUSSION TITLE
implementing "ex af,af'" for a z80 version

### DISCUSSION THREAD

**Discussion by physnoct** [hlorenzi/customasm#147] [Q&A]
I have done a draft of a Z80 definition for use with customASM.
It seems I can't use a ' in a definition.

Here is my definition for this instruction:
#ruledef
{
	ex af,af'	=> 0x08
}

And the result:
customasm v0.11.14-1-gcc05721 (2022-01-30, x86_64-unknown-linux-gnu)
assembling `test_cpu.asm`...

error: unexpected character
 --> test_cpu.asm:219:10:
 217 |  
 218 |   EX  DE,HL 
 219 |   EX  AF,AF' 
     |            ^ 

How can we solve this?

**Comment by hlorenzi** [comment]
Oh! I just haven't set apostrophes to be allowed in patterns. I think there should be no problem in allowing it in the next version. Could you please create an issue for this?

Out of curiosity, is that standard syntax for the Z80?

**Reply by mainrs** [reply]
Yes, they syntax is used in multiple websites that provide instruction sets for the z80. 

https://spectrumforeveryone.com/technical/z80-processor-instructions/
https://retro-computers.fandom.com/wiki/Z80_Instruction_Set
http://www.z80.info/z80-op.txt
